CMakeLists.txt

```diff
@@ -65,11 +67,16 @@ endif()
 
 message("python=${PYTHON_EXECUTABLE}")
 
-add_custom_target(
+if (TI_GENERATE_COMMIT_HASH)
+  add_custom_target(
     generate_commit_hash
     COMMAND ${PYTHON_EXECUTABLE} "${CMAKE_CURRENT_LIST_DIR}/misc/generate_commit_hash.py"
-)
-add_dependencies(${CORE_LIBRARY_NAME} generate_commit_hash)
+    )
+  add_dependencies(${CORE_LIBRARY_NAME} generate_commit_hash)
+else()
+  FILE(WRITE ${CMAKE_CURRENT_LIST_DIR}/taichi/common/commit_hash.h
+    "#define TI_COMMIT_HASH \"${TI_COMMIT_HASH}\"\n")
```

Comment:
I found we already have `${TI_COMMIT_HASH}` in L24, so I used it for TI_GENERATE_COMMIT_HASH=OFF.